src/rust/ensogl/lib/core/src/gui/component.rs

```diff
@@ -189,6 +189,61 @@ impl<T> display::Object for ShapeView<T> {
 
 
 
+// ==================
+// === Animatable ===
+// ==================
+
+/// Indicate what datatype to use in the animation space representation.
+pub trait HasAnimationSpaceRepr {
+    /// Representation in animation space. Needs to support linear interpolation and all
+    /// pre-requisites of `inertia::Value`.
+    type AnimationSpaceRepr: inertia::Value;
+}
+
+/// Newtype that indicates that the wrapped value is valid to be used in animations.
+#[derive(Debug)]
+pub struct AnimationLinearSpace<T> {
+    /// Wrapped value representing the animation space value.
+    pub value: T
+}
+
+/// Shorthand for a  HasAnimationSpaceRepr::AnimationSpaceRepr wrapped in a `AnimationLinearSpace`.
+pub type AnimationSpaceRepr<T> = AnimationLinearSpace<<T as HasAnimationSpaceRepr>::AnimationSpaceRepr>;
+
+/// This is used to make type inference better on usage places - without it,
+/// Rust would force one to write the where part EVERYWHERE in use places.
+pub trait FromAsInto<T> = Sized where T:Into<Self>;
+
+/// Can be transformed from and into.
+pub trait FromInto<T> = Sized + Into<T> + FromAsInto<T>;
```

Comment:
Why not `Iso`? This is a standard form of isomorphism and imo this sohuld be named this way. Moreover, why not placing it in prelude like I proposed in my snippet?

Reply:
We don't really define a mathematical isomorphism here. That is [a lot more specific](https://en.wikipedia.org/wiki/Isomorphism). All we do require is a `From<T>` and `Into<T>` implementation. From a Rust perspective, the name `FromInto` is much clearer about the requirements. 

I agree, needs to move to prelude though.

Reply:
> I agree, needs to move to prelude though.

Just noticed prelude is in a different crate, though I would leave this here for now. If we feel strongly about the usefulness, we can open an issue to add it there and refactor here after it has landed.

Reply:
The fact that prelude is in different crate should not prevent us from changing it. This should definitely be in prelude IMO. If youre not sure, lets talk about it. Also, lets talk more aobut the Iso name, as I believe this would be the correct name for that.